libavfilter/dnn/dnn_backend_native_layer_conv2d_x86.asm

```diff
@@ -0,0 +1,214 @@
+;* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+;* Lesser General Public License for more details.
+;*
+;* You should have received a copy of the GNU Lesser General Public
+;* License along with FFmpeg; if not, write to the Free Software
+;* Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+;******************************************************************************
+
+%include "libavutil/x86/x86util.asm"
+
+SECTION .text
+
+%macro COUNT_INPUT 0
+    mov tmp1d, padding_method
+    cmp tmp1d, SAME_CLAMP_TO_EDGE
+    je .clamp
+
+    cmp y_posd, 0
+    jl .out_of_th
+    mov tmp2d, height
+    cmp y_posd, tmp2d
+    jge .out_of_th
+
+    cmp x_posd, 0
+    jl .out_of_th
+    mov tmp2d, width
+    cmp x_posd, tmp2d
+    jge .out_of_th
+
+    mov tmp1d, y_posd
+    imul tmp1d, src_linesize
+    mov tmp2d, x_posd
+    imul tmp2d, input_num
+    add tmp1d, tmp2d
+    jmp .count_end
+
+    .out_of_th:
+        mov tmp1d, -1
+        jmp .count_end
+
+    .clamp:
+    cmp y_posd, 0
+    jl .y_clamp_zero
+    mov tmp1d, height
+    cmp y_posd, tmp1d
+    jge .y_clamp_height
+    mov tmp1d, y_posd
+    jmp .y_normal
+
+    .y_clamp_zero:
+        xor tmp1d, tmp1d
+        jmp .y_normal
+
+    .y_clamp_height:
+        sub tmp1d, 1
+
+    .y_normal:
+
+    cmp x_posd, 0
+    jl .x_clamp_zero
+    mov tmp2d, width
+    cmp x_posd, tmp2d
+    jge .x_clamp_width
+    mov tmp2d, x_posd
+    jmp .x_normal
+
+    .x_clamp_zero:
+        xor tmp2d, tmp2d
+        jmp .x_normal
+
+    .x_clamp_width:
+        sub tmp2d, 1
+
+    .x_normal:
+
+    imul tmp1d, src_linesize
+    imul tmp2d, input_num
+    add tmp1d, tmp2d
+
+    .count_end:
+%endmacro
+
+; void ff_dnn_execute_layer_conv2d_sse4(const execute_param *exe_param);
+
+%if ARCH_X86_64
+INIT_XMM sse4
+cglobal dnn_execute_layer_conv2d, 8, 15, 3, exe_param,\
+    x, y, n_filter, cha, kernel_x, kernel_y, x_pos, y_pos, kernel_pos,\
+    input, output, kernel, tmp1, tmp2
+
+%define thread_start [exe_paramq]
+%define thread_end [exe_paramq + 1 * 4]
+%define input_num [exe_paramq + 2 * 4]
+%define output_num [exe_paramq + 3 * 4]
+%define kernel_size [exe_paramq + 4 * 4]
+%define padding_method [exe_paramq + 5 * 4]
+%define dilation [exe_paramq + 6 * 4]
+%define pad_size [exe_paramq + 7 * 4]
+%define width [exe_paramq + 8 * 4]
+%define height [exe_paramq + 9 * 4]
+%define radius [exe_paramq + 10 * 4]
+%define src_linesize [exe_paramq + 11 * 4]
+%define filter_size [exe_paramq + 12 * 4]
+%define filter_linesize [exe_paramq + 13 * 4]
+%define SAME_CLAMP_TO_EDGE 2
+
+    mov inputq, [exe_paramq + 14 * 4]
+    mov outputq, [exe_paramq + 14 * 4 + 8]
+    mov kernelq, [exe_paramq + 14 * 4 + 2 * 8]
+
+    mov yd, thread_start
+.loop_y:
+    mov xd, pad_size
+    .loop_x:
+        xor n_filterd, n_filterd
+        xor kernel_posq, kernel_posq
+        .loop_filter:
+            xorps m2, m2
+            xor kernel_yd, kernel_yd
+
```

Comment:
why empty line here and there? 

better to add comments for a different logic block.